sdk/src/dlob/DLOB.ts

```diff
@@ -568,63 +648,29 @@ export class DLOB {
 		oraclePriceData: OraclePriceData,
 		slot: number
 	): {
-		crossingNodes?: NodeToFill;
-		exhaustedSide?: Side;
+		crossingNodes: NodeToFill[];
+		exhaustedSide: Side;
 	} {
 		const bidPrice = bidNode.getPrice(oraclePriceData, slot);
 		const askPrice = askNode.getPrice(oraclePriceData, slot);
 
 		const bidOrder = bidNode.order;
 		const askOrder = askNode.order;
 
-		const containsMarketOrder =
-			(askOrder && isVariant(askOrder.orderType, 'market')) ||
-			(bidOrder && isVariant(bidOrder.orderType, 'market'));
-
-		if (!containsMarketOrder || bidPrice.lt(askPrice)) {
-			// no market orders, and no crossing orders - return
-			if (askNode.isVammNode() && !bidNode.isVammNode()) {
-				return {
-					exhaustedSide: 'bid',
-				};
-			} else if (!askNode.isVammNode() && bidNode.isVammNode()) {
-				return {
-					exhaustedSide: 'ask',
-				};
-			} else {
-				return {
-					exhaustedSide: 'both',
-				};
-			}
-		}
-
-		// User bid crosses the vamm ask
-		if (askNode.isVammNode()) {
-			if (!isAuctionComplete(bidOrder, slot)) {
-				return {
-					exhaustedSide: 'bid',
-				};
-			}
+		// Can't match two maker orders or if maker and taker are the same
+		const makerIsTaker = bidNode.userAccount.equals(askNode.userAccount);
+		const ordersDontCross = bidPrice.lt(askPrice);
```

Comment:
if the orders dont cross why wouldnt we want to exhaust both sides?

Reply:
oh that is a good point 😅

Reply:
actually if we ever hit two nodes that don't cross, we shouldn't find two nodes that cross even if we continue to walk up the book